tests/sklearn/test_sklearn_models.py

```diff
@@ -1139,6 +1140,38 @@ def check_load_fitted_sklearn_linear_models(model_class, n_bits, x, y, check_flo
     )
 
 
+def check_rounding_consistency(
+    model,
+    x,
+    predict_method,
+    metric,
+):
+
+    """Test that Concrete ML witout and with rounding are 'equivalent'."""
+
+    random_int = random.randint(0, x.shape[0] - 1)
+
+    model.compile(x)
+
+    rounded_predict_quantized = predict_method(x, fhe="disable")
+    rounded_predict_simulate = predict_method(x, fhe="simulate")
+    rounded_predict_fhe = predict_method(x[random_int, None], fhe="execute")
+
+    # pylint: disable=protected-access
+    quant_x = model.quantize_input(x).astype("float")
+    model._convert_tree_to_numpy_and_compute_lsbs_to_remove(quant_x, use_rounding=False)
```

Comment:
I agree that we could have a more "hidden" flag for rounding/no-rounding. But I think we should also have correctness test between roudning/no-rounding. As it happens trees are "correct" w.r.t. scikit-learn , so we could, as you say, test directly that correctness. But at some point trees might change w.r.t scikilearn (PTQ for example) and then we could have problems.

Reply:
I'm not sure I get it why this test would be need in case of having PTQ 🤔 and even, I would add the test when it's needed in that case no ? 

I'm also afraid that this would add a new flaky test, which could even be non-reproductible as @kcelia mentioned it in recent weeklys !